src/gtest/test_noteencryption.cpp

```diff
@@ -78,6 +78,35 @@ TEST(noteencryption, sapling_api)
         small_message
     );
 
+    // Test nonce-reuse resistance of API
+    {
+        auto tmp_enc = *SaplingNoteEncryption::FromDiversifier(pk_1.d);
+        
+        tmp_enc.encrypt_to_recipient(
+            pk_1.pk_d,
+            message
+        );
+
+        ASSERT_THROW(tmp_enc.encrypt_to_recipient(
+            pk_1.pk_d,
```

Comment:
If we change this line to `pk_2.pk_d` the call still throws the same error due to `already_encrypted_enc` having been set, so we're not really testing nonce reuse, but the fact that `encrypt_to_recipient` can be only called once successfully.

Reply:
That's the intent of the nonce reuse resistant API; if you call `encrypt_to_recipient` twice (which you should never do, ephemeral keys are not reused) then the ChaCha20 nonce will be reused, which we cannot allow.

Reply:
Ok.  Note that `encrypt_to_recipient` could be called twice because the method is not thread safe with regards to the boolean flag.  Perhaps document that the note encryption class should not be used by multiple threads concurrently.